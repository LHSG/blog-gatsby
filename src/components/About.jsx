/* eslint-disable react/no-danger */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Image from './Image';
import { GitHub, Keybase, LinkedIn, Twitter } from './icons';
import { siteSocial, image as imagePropType } from '../proptypes';

const Title = styled.h1`
  margin-bottom: 2.175rem;
`;

const SocialLinks = styled.div`
  text-align: right;
`;

const SocialLink = styled.a`
  display: inline-block;
  background-image: none;

  &:hover {
    background-image: none;
  }
`;

export default function About({
  title,
  html,
  social: { githubUrl, keybaseUrl, linkedInUrl, twitterUrl },
  coverImageSizes,
}) {
  return (
    <div>
      <Title>{title}</Title>
      <Image alt={title} sizes={coverImageSizes} />
      <div dangerouslySetInnerHTML={{ __html: html }} />
      <SocialLinks>
        <SocialLink
          href={githubUrl}
          title="My GitHub Profile"
          target="_blank"
          rel="noopener noreferrer"
        >
          <GitHub enlargeOnMobile />
        </SocialLink>
        <SocialLink
          href={keybaseUrl}
          title="My Keybase Profile"
          target="_blank"
          rel="noopener noreferrer"
        >
          <Keybase enlargeOnMobile />
        </SocialLink>
        <SocialLink
          href={twitterUrl}
          title="My Twitter Profile"
          target="_blank"
          rel="noopener noreferrer"
        >
          <Twitter enlargeOnMobile />
        </SocialLink>
        <SocialLink
          href={linkedInUrl}
          title="My LinkedIn Profile"
          target="_blank"
          rel="noopener noreferrer"
        >
          <LinkedIn enlargeOnMobile />
        </SocialLink>
      </SocialLinks>
    </div>
  );
}

About.propTypes = {
  html: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  social: siteSocial.isRequired,
  coverImageSizes: imagePropType.isRequired,
};
