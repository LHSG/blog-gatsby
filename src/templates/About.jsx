import React from 'react';
import PropTypes from 'prop-types';

import { Head, About } from '../components';
import {
  markdownRemark as markdownRemarkPropType,
  site as sitePropType,
} from '../proptypes';

export default function AboutTemplate({
  data: {
    site,
    markdownRemark: {
      frontmatter: {
        title,
        path,
        cover: { childImageSharp: { sizes: coverImageSizes } },
      },
      html,
    },
  },
}) {
  return (
    <div>
      <Head title={title} path={path} site={site} />
      <About
        title={title}
        html={html}
        social={site.siteMetadata.social}
        coverImageSizes={coverImageSizes}
      />
    </div>
  );
}

AboutTemplate.propTypes = {
  data: PropTypes.shape({
    markdownRemark: markdownRemarkPropType,
    site: sitePropType.isRequired,
  }).isRequired,
};

export const pageQuery = graphql`
  query AboutPage($path: String!) {
    ...SiteFragment
    markdownRemark(frontmatter: { path: { eq: $path } }) {
      ...MarkdownMetadataFragment
      ...MarkdownFrontmatterWithCoverFragment
    }
  }
`;
