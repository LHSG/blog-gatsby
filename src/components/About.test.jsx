import React from 'react';
import { shallow } from 'enzyme';

import About from './About';
import { imagePropType } from '../../test/fixtures';

test('rendering', () => {
  const props = {
    title: 'title',
    html: '<div>hello</div>',
    social: {
      githubUrl: 'githubUrl',
      keybaseUrl: 'keybaseUrl',
      linkedInUrl: 'linkedInUrl',
      twitterUrl: 'twitterUrl',
    },
    coverImageSizes: imagePropType,
  };
  const wrapper = shallow(<About {...props} />);
  expect(wrapper).toMatchSnapshot();
});
