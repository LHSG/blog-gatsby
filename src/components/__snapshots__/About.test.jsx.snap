// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<div>
  <styled.h1>
    title
  </styled.h1>
  <Styled(Image)
    alt="title"
    sizes={
      Object {
        "sizes": "something",
      }
    }
  />
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>hello</div>",
      }
    }
  />
  <styled.div>
    <styled.a
      href="githubUrl"
      rel="noopener noreferrer"
      target="_blank"
      title="My GitHub Profile"
    >
      <GithubIcon
        enlargeOnMobile={true}
      />
    </styled.a>
    <styled.a
      href="keybaseUrl"
      rel="noopener noreferrer"
      target="_blank"
      title="My Keybase Profile"
    >
      <KeybaseIcon
        enlargeOnMobile={true}
      />
    </styled.a>
    <styled.a
      href="twitterUrl"
      rel="noopener noreferrer"
      target="_blank"
      title="My Twitter Profile"
    >
      <TwitterIcon
        enlargeOnMobile={true}
      />
    </styled.a>
    <styled.a
      href="linkedInUrl"
      rel="noopener noreferrer"
      target="_blank"
      title="My LinkedIn Profile"
    >
      <LinkedIn
        enlargeOnMobile={true}
      />
    </styled.a>
  </styled.div>
</div>
`;
